plugin/evm/atomic/vm/export_tx_test.go

```diff
@@ -856,102 +837,89 @@ func TestExportTxSemanticVerify(t *testing.T) {
 		},
 		{
 			name: "too few signatures",
-			tx:   &atomic.Tx{UnsignedAtomicTx: validExportTx},
+			tx:   validExportTx,
 			signers: [][]*secp256k1.PrivateKey{
 				{key},
 				{key},
 			},
-			baseFee:   vmtest.InitialBaseFee,
-			rules:     vmtest.ForkToRules(upgradetest.ApricotPhase3),
-			shouldErr: true,
+			fork:    upgradetest.ApricotPhase3,
+			wantErr: errIncorrectNumCredentials,
 		},
 		{
 			name: "too many signatures on credential",
-			tx:   &atomic.Tx{UnsignedAtomicTx: validExportTx},
+			tx:   validExportTx,
 			signers: [][]*secp256k1.PrivateKey{
 				{key, vmtest.TestKeys[1]},
 				{key},
 				{key},
 			},
-			baseFee:   vmtest.InitialBaseFee,
-			rules:     vmtest.ForkToRules(upgradetest.ApricotPhase3),
-			shouldErr: true,
+			fork:    upgradetest.ApricotPhase3,
+			wantErr: errIncorrectNumSignatures,
 		},
 		{
 			name: "too few signatures on credential",
-			tx:   &atomic.Tx{UnsignedAtomicTx: validExportTx},
+			tx:   validExportTx,
 			signers: [][]*secp256k1.PrivateKey{
 				{},
 				{key},
 				{key},
 			},
-			baseFee:   vmtest.InitialBaseFee,
-			rules:     vmtest.ForkToRules(upgradetest.ApricotPhase3),
-			shouldErr: true,
+			fork:    upgradetest.ApricotPhase3,
+			wantErr: errIncorrectNumSignatures,
 		},
 		{
 			name: "wrong signature on credential",
-			tx:   &atomic.Tx{UnsignedAtomicTx: validExportTx},
+			tx:   validExportTx,
 			signers: [][]*secp256k1.PrivateKey{
 				{vmtest.TestKeys[1]},
 				{key},
 				{key},
 			},
-			baseFee:   vmtest.InitialBaseFee,
-			rules:     vmtest.ForkToRules(upgradetest.ApricotPhase3),
-			shouldErr: true,
+			fork:    upgradetest.ApricotPhase3,
+			wantErr: errPublicKeySignatureMismatch,
 		},
 		{
-			name:      "no signatures",
-			tx:        &atomic.Tx{UnsignedAtomicTx: validExportTx},
-			signers:   [][]*secp256k1.PrivateKey{},
-			baseFee:   vmtest.InitialBaseFee,
-			rules:     vmtest.ForkToRules(upgradetest.ApricotPhase3),
-			shouldErr: true,
+			name:    "no signatures",
+			tx:      validExportTx,
+			signers: [][]*secp256k1.PrivateKey{},
+			fork:    upgradetest.ApricotPhase3,
+			wantErr: errIncorrectNumCredentials,
 		},
 	}
 	for _, test := range tests {
-		if err := test.tx.Sign(atomic.Codec, test.signers); err != nil {
-			t.Fatal(err)
-		}
+		t.Run(test.name, func(t *testing.T) {
+			tx := &atomic.Tx{UnsignedAtomicTx: test.tx}
+			require.NoError(t, tx.Sign(atomic.Codec, test.signers))
 
-		backend := NewVerifierBackend(vm, *test.rules)
+			rules := vmtest.ForkToRules(test.fork)
+			backend := NewVerifierBackend(vm, *rules)
 
-		t.Run(test.name, func(t *testing.T) {
-			tx := test.tx
-			err := backend.SemanticVerify(tx, parent, test.baseFee)
-			if test.shouldErr && err == nil {
-				t.Fatalf("should have errored but returned valid")
-			}
-			if !test.shouldErr && err != nil {
-				t.Fatalf("shouldn't have errored but returned %s", err)
-			}
+			err := backend.SemanticVerify(tx, parent, vmtest.InitialBaseFee)
+			require.ErrorIs(t, err, test.wantErr)
```

Comment:
Isn't there a linter that requires this to be in-lined? my mind has been consumed by the linters...